trailblazer/services/slurm/slurm_api_service/slurm_api_service.py

```diff
@@ -0,0 +1,24 @@
+from trailblazer.clients.slurm_api_client.dto.job_response import SlurmJobResponse
+from trailblazer.clients.slurm_api_client.slurm_api_client import SlurmAPIClient
+from trailblazer.services.slurm.dtos import SlurmJobInfo
+from trailblazer.services.slurm.slurm_api_service.utils import create_job_info_dto
+from trailblazer.services.slurm.slurm_service import SlurmService
+
+
+class SlurmAPIService(SlurmService):
+    def __init__(self, client: SlurmAPIClient):
+        self.client = client
+
+    def get_job(self, job_id: int) -> SlurmJobInfo | None:
+        job_response: SlurmJobResponse = self.client.get_job(job_id)
+        if not job_response.jobs or job_response.errors:
+            return None
```

Comment:
Are we leaving error handling for the future?

Reply:
Good point, to ensure the behavior that we have for the `SlurmCLIService` we should raise an exception the client instead for this. The existing behaviour is that, if a single job cannot be updated for an analysis, the entire status of the analysis is set to error.

Reply:
This already happens. Removed the unnecessary check and added some additional error handling in https://github.com/Clinical-Genomics/trailblazer/pull/451/commits/3d452a77bff25502e071e61a96d1af9112906dbb